# Terragrunt fails to copy files if an SSH agent is running on my system.

**dpedu** commented *Feb 11, 2020*

I'm setting up a basic test using Terratest that eventually hits what appears to be a Terragrunt bug. My test looks like this:

```go
func TestBasic(t *testing.T) {
	testFolder, err := files.CopyTerragruntFolderToTemp("./", t.Name())
	require.NoError(t, err)

	options := &terraform.Options{
		TerraformDir:    testFolder + "/vpc",
		TerraformBinary: "terragrunt",
		Vars: map[string]interface{}{
			"vpc_name": "foobar",
		},
	}

	defer terraform.Destroy(t, options)
	terraform.InitAndApply(t, options)
}
```

This go file lives in my project at a path like `basedir/test/testname/foo_test.go`. In the same directory is the "main" terragrunt.hcl file, a yaml file, and a subdirectory (named `vpc` and referenced above) containing another terragrunt.hcl pointing at some module. I intend this test to copy this entire tree somewhere else and run terragrunt. I'm going to add dependencies later.

Anyway, when I run this test, using `go test -v -timeout 90m -count=1  -run TestBasic`, I get the following output:

```
TestBasic 2020-02-11T14:14:24-08:00 command.go:87: Running command terragrunt with args [init -upgrade=false --terragrunt-non-interactive]
TestBasic 2020-02-11T14:14:25-08:00 command.go:158: [terragrunt] 2020/02/11 14:14:25 Reading Terragrunt config file at /private/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T/TestBasic464735386/basic/vpc/terragrunt.hcl
TestBasic 2020-02-11T14:14:25-08:00 command.go:158: [terragrunt] [/private/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T/TestBasic464735386/basic/vpc] 2020/02/11 14:14:25 Running command: terraform --version
TestBasic 2020-02-11T14:14:25-08:00 command.go:158: [terragrunt] 2020/02/11 14:14:25 Downloading Terraform configurations from file:///private/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T into /private/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T/TestBasic464735386/basic/vpc/.terragrunt-cache/k22W8QiNhJo9mWaEGK07ZTQU88E/9bvt-VdsXERdCqSKNpUAivsqAAg
TestBasic 2020-02-11T14:14:25-08:00 command.go:158: [terragrunt] 2020/02/11 14:14:25 Hit multiple errors:
TestBasic 2020-02-11T14:14:25-08:00 command.go:158: error downloading 'file:///private/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T': open /private/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T/ssh-HA4LT4RImhn6/agent.501: operation not supported on socket
TestBasic 2020-02-11T14:14:25-08:00 command.go:158: [terragrunt] 2020/02/11 14:14:25 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
TestBasic 2020-02-11T14:14:25-08:00 retry.go:80: Returning due to fatal error: FatalError{Underlying: exit status 1}
```

This seems to be the problem in particular: `command.go:158: error downloading 'file:///private/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T': open /private/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T/ssh-HA4LT4RImhn6/agent.501: operation not supported on socket`. The temporary path in use (produced by the print statement in the test) is something like `/var/folders/ms/46n9q4gs0j575n1nbgk7kff40000gp/T/TestBasic464735386/basic`.

The path in the error message is a unix socket owned by my SSH agent. Why is Terragrunt messing with my SSH agent?

This can be worked around by setting `TMPDIR=/some/other/path` which causes Terratest to create tmpdirs somewhere else.

Edit: I believe I caused this by not having the right files for Terragrunt in the destination. With a TMPDIR elsewhere, it complained about not being able to find main.tf, which is what is was probably searching for. Closing this.
<br />
***


